tests/src/test/scala/mhtml/tests.scala

```diff
@@ -410,4 +406,41 @@ class Tests extends FunSuite {
     assert(d.levenshtein("kitten")("sitting") == 3)
     assert(d.levenshtein("rosettacode")("raisethysword") == 8)
   }
+
+  test("Optional Attribute") {
+    assert(render(<form disabled={Rx(true)}></form>) == """<form disabled=""></form>""")
+    assert(render(<form disabled={Rx(false)}></form>) == """<form></form>""")
+    assert(render(<form disabled={true}></form>) == """<form disabled=""></form>""")
+    assert(render(<form disabled={false}></form>) == """<form></form>""")
+    assert(render(<form disabled={Option("banana")}></form>) == """<form disabled="banana"></form>""")
+    assert(render(<form disabled={Option.empty[String]}></form>) == """<form></form>""")
+  }
+
+  test("Ill-typed Attribute") {
+    assertTypeError("<form disabled={Option.empty[Int]}></form>")
+    assertTypeError("<form disabled={List.empty[String]}></form>")
+    assertTypeError("<form disabled={List.empty[Text]}></form>")
+    assertTypeError("<form disabled={Text(\"\"): xml.Node}></form>")
+    assertTypeError("<form disabled={Rx(<div></div>)}></form>")
+  }
+
+  test("ill-typed Element") {
+    assertTypeError("<div>{List.empty[String]}</div>")
+    assertTypeError("{ class A; <div>{new A}</div>}")
+  }
+
+  test("well-typed Element") {
+    assert(render(<div>{Rx(Option.empty[Int])}</div>) == "<div></div>")
+    assert(render(<div>{Option.empty[Int]}</div>) == "<div></div>")
+    assert(render(<div>{Rx(Option(1))}</div>) == "<div>1</div>")
+    assert(render(<div>{Option(1)}</div>) == "<div>1</div>")
+    assert(render(<div>{1}</div>) == "<div>1</div>")
+    assert(render(<div>{2L}</div>) == "<div>2</div>")
+    assert(render(<div>{1.2d}</div>) == "<div>1.2</div>")
+    assert(render(<div>{1.0f}</div>) == "<div>1</div>")
+    assert(render(<div>{"string"}</div>) == "<div>string</div>")
+    assert(render(<div>{List(scala.xml.Comment("a"))}</div>) == "<div><!--a--></div>")
+    assert(render(<div>{List(scala.xml.Comment("a"))}</div>) == "<div><!--a--></div>")
```

Comment:
Good catch. I've refactored these tests into a separate test suite so that each assertion becomes a test with the name of the expected output. This should help catch duplicate tests in the future.